code/modules/client/preferences_savefile.dm

```diff
@@ -141,50 +139,49 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 		var/datum/keybinding/conflicted = item
 		to_chat(parent, span_danger("[conflicted.category]: [conflicted.full_name] needs updating"))
 
-
-/datum/preferences/proc/load_path(ckey,filename="preferences.sav")
+/datum/preferences/proc/load_path(ckey, filename="preferences.json")
 	if(!ckey)
 		return
-	path = "data/player_saves/[ckey[1]]/[ckey]/[filename]"
+	// path = "data/player_saves/[ckey[1]]/[ckey]/[filename]"
+	path = "data/player_saves_json_testing/[ckey[1]]/[ckey]/[filename]"
 
-/datum/preferences/proc/load_preferences()
+/datum/preferences/proc/load_savefile()
 	if(!path)
-		return FALSE
-	if(!fexists(path))
-		return FALSE
+		CRASH("Attempted to load savefile without first loading a path!")
+	savefile = new /datum/json_savefile(path)
 
-	var/savefile/S = new /savefile(path)
-	if(!S)
+/datum/preferences/proc/load_preferences()
+	if(!savefile)
+		stack_trace("Attempted to load the preferences of [parent] without a savefile; did you forget to call load_savefile?")
+		load_savefile()
+	if(!savefile)
+		stack_trace("Failed to load the savefile for [parent] after manually calling load_savefile; something is very wrong.")
 		return FALSE
```

Comment:
```suggestion
	if(!savefile)
		stack_trace("Attempted to load the preferences of [parent] without a savefile; did you forget to call load_savefile?")
		load_savefile()
		if(!savefile)
			stack_trace("Failed to load the savefile for [parent] after manually calling load_savefile; something is very wrong.")
			return FALSE
```

I don't see why we have to evaluate if the savefile is null twice if we succeed the first time.